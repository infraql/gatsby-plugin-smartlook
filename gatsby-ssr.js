"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.onRenderBody = void 0;

var _react = _interopRequireDefault(require("react"));

var onRenderBody = function onRenderBody(_ref, pluginOptions) {
  var setHeadComponents = _ref.setHeadComponents;

  if (process.env.NODE_ENV === "production") {
    return setHeadComponents([/*#__PURE__*/_react.default.createElement("script", {
      key: "gatsby-plugin-smartlook",
      dangerouslySetInnerHTML: {
        __html: "\nwindow.smartlook||(function(d) {\nvar o=smartlook=function(){ o.api.push(arguments)},h=d.getElementsByTagName('head')[0];\nvar c=d.createElement('script');o.api=new Array();c.async=true;c.type='text/javascript';\nc.charset='utf-8';c.src='https://rec.smartlook.com/recorder.js';h.appendChild(c);\n})(document);\nsmartlook('init', '" + pluginOptions.projectKey + "');\n      "
      }
    })]);
  }

  return null;
};

exports.onRenderBody = onRenderBody;