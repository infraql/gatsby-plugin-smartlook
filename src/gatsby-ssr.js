import React from "react"

export const onRenderBody = ({ setHeadComponents }, pluginOptions) => {
  if (process.env.NODE_ENV === `production`) {
    return setHeadComponents([
      <script
        key={`gatsby-plugin-smartlook`}
        dangerouslySetInnerHTML={{
          __html: `
window.smartlook||(function(d) {
var o=smartlook=function(){ o.api.push(arguments)},h=d.getElementsByTagName('head')[0];
var c=d.createElement('script');o.api=new Array();c.async=true;c.type='text/javascript';
c.charset='utf-8';c.src='https://rec.smartlook.com/recorder.js';h.appendChild(c);
})(document);
smartlook('init', '${pluginOptions.projectKey}');
      `,
        }}
      />,
    ])
  }

  return null
}