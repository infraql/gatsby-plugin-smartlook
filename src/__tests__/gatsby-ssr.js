import { onRenderBody } from "../gatsby-ssr"

describe(`gatsby-plugin-smartlook`, () => {
  describe(`onRenderBody`, () => {
    describe(`in development mode`, () => {
      it(`does not set any head components`, () => {
        const setHeadComponents = jest.fn()

        onRenderBody({ setHeadComponents }, {})

        expect(setHeadComponents).not.toHaveBeenCalled()
      })
    })

    describe(`in production mode`, () => {
      let env

      beforeAll(() => {
        env = process.env.NODE_ENV
        process.env.NODE_ENV = `production`
      })

      afterAll(() => {
        process.env.NODE_ENV = env
      })

      it(`sets the correct head components`, () => {
        const setHeadComponents = jest.fn()
        const pluginOptions = { projectKey: `12345` }

        onRenderBody({ setHeadComponents }, pluginOptions)

        expect(setHeadComponents.mock.calls).toMatchSnapshot()
      })
    })
  })
})