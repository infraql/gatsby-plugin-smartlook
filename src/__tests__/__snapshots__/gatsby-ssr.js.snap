// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-smartlook onRenderBody in production mode sets the correct head components 1`] = `
Array [
  Array [
    Array [
      <script
        dangerouslySetInnerHTML={
          Object {
            "__html": "
window.smartlook||(function(d) {
var o=smartlook=function(){ o.api.push(arguments)},h=d.getElementsByTagName('head')[0];
var c=d.createElement('script');o.api=new Array();c.async=true;c.type='text/javascript';
c.charset='utf-8';c.src='https://rec.smartlook.com/recorder.js';h.appendChild(c);
})(document);
smartlook('init', '12345');
      ",
          }
        }
      />,
    ],
  ],
]
`;